packages/backend/src/App.ts

```diff
@@ -374,67 +374,77 @@ export default class App {
-                        this.lightdashConfig.mode,
-                    )
-                        ? 'unsafe-none'
-                        : 'same-origin',
+        const helmetConfig = {
+            contentSecurityPolicy: {
+                directives: {
+                    'default-src': [
+                        "'self'",
+                        ...contentSecurityPolicyAllowedDomains,
+                    ],
+                    'img-src': ["'self'", 'data:', 'https://*'],
+                    'frame-src': ["'self'", 'https://*'],
+                    'frame-ancestors': [
+                        "'self'",
+                        ...this.lightdashConfig.security.contentSecurityPolicy
+                            .frameAncestors,
+                    ],
+                    'worker-src': [
+                        "'self'",
+                        'blob:',
+                        ...contentSecurityPolicyAllowedDomains,
+                    ],
+                    'child-src': [
+                        // Fallback of worker-src for safari older than 15.5
+                        "'self'",
+                        'blob:',
+                        ...contentSecurityPolicyAllowedDomains,
+                    ],
+                    'script-src': [
+                        "'self'",
+                        "'unsafe-eval'",
+                        ...contentSecurityPolicyAllowedDomains,
+                    ],
+                    'script-src-elem': [
+                        "'self'",
+                        "'unsafe-inline'",
+                        ...contentSecurityPolicyAllowedDomains,
+                    ],
+                    'report-uri': reportUris.map((uri) => uri.href),
                 },
-            }),
-        );
+                reportOnly:
+                    this.lightdashConfig.security.contentSecurityPolicy
+                        .reportOnly,
+            },
+            strictTransportSecurity: {
+                maxAge: 31536000,
+                includeSubDomains: true,
+                preload: true,
+            },
+            referrerPolicy: {
+                policy: 'strict-origin-when-cross-origin',
+            },
+            noSniff: true,
+            xFrameOptions: false,
+            crossOriginOpenerPolicy: {
+                policy: [LightdashMode.DEMO, LightdashMode.PR].includes(
+                    this.lightdashConfig.mode,
+                )
+                    ? 'unsafe-none'
+                    : 'same-origin',
+            },
+        } as const;
+
+        expressApp.use(helmet(helmetConfig));
+
+        const helmetConfigForEmbeds = produce(helmetConfig, (draft) => {
+            // eslint-disable-next-line no-param-reassign
+            draft.contentSecurityPolicy.directives['frame-ancestors'] = [
+                "'self'",
+                'https://*',
+            ];
+        });
+
+        expressApp.use('/embed/*', helmet(helmetConfigForEmbeds));
```

Comment:
don't we need to give access also to the entire app and explore pages ?. Can we just limit /login /register pages instead?